cdk/multiplayer_test_scaler/server_artifacts_automation.py

```diff
@@ -0,0 +1,144 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT-0
+
+from aws_cdk import (
+    Stack,
+    aws_events as events,
+    aws_events_targets as events_targets,
+    aws_iam as iam,
+    aws_lambda as _lambda,
+    aws_ssm as ssm,
+)
+import aws_cdk as cdk
+from constructs import Construct
+
+from .constants import *
+
+class ServerAutomationConstruct(Construct):
+    """
+    Builds automation resources for pulling logs and metrics files off the server
+    """
+
+    def __init__(self, scope: Construct, construct_id: str) -> None:
+        super().__init__(scope, construct_id)
+        
+    def create_file_sync_rule(self, artifact_bucket_name: str) -> events.CfnRule:
+        self._create_command_document(artifact_bucket_name)
+
+        doc_arn = Stack.of(self).format_arn(
+            service="ssm",
+            resource="document",
+            resource_name=self._file_sync_document.name
+        )
+
+        doc_run_role = iam.Role(
+            self, f'{RESOURCE_ID_COMMON_PREFIX}InvokeUploadRuleRole',
+            description='Role to be used by Event Bridge to run file sync rule',
+            assumed_by=iam.ServicePrincipal('events.amazonaws.com'),
+            inline_policies={'InvokeFileSyncRulePolicy': iam.PolicyDocument(
+                statements=[
+                    iam.PolicyStatement(
+                        effect=iam.Effect.ALLOW,
+                        actions=['ssm:SendCommand'],
+                        resources=[
+                            doc_arn,
+                            cdk.Fn.sub('arn:${AWS::Partition}:ec2:${AWS::Region}:${AWS::AccountId}:instance/*'),
+                        ]
+                    )
+                ])
+            }
+        )
+
+        # can't use L2 events.Rule b/c it doesn't yet support
+        # SSM RunCommand targets. See: https://github.com/aws/aws-cdk/issues/7710
+        self._file_sync_rule = events.CfnRule(self, 'IntermitentUploadRule',
+            name=f'{RESOURCE_ID_COMMON_PREFIX}SyncArtifactsRule',
+            schedule_expression='rate(2 minutes)',
+            targets=[events.CfnRule.TargetProperty(
+                arn=doc_arn,
+                id='MpScalerFileSyncRule',
```

Comment:
nit: MpsScaler

Reply:
the headline title of the tool is "Multiplayer Test Scaler", so I think "MP Scaler" is the better abbreviation. I know historically we've referred to it as "Multiplayer sample scaler" to refer to the main test case (O3DE Multiplayer Sample), but the intention is to be more generically useful so I'd rather drop the "sample"